### DISCUSSION TITLE
Setting TX Power and Spreadingfactor dynamicly

### DISCUSSION THREAD

**Discussion by oedze** [Lora-net/LoRaMac-node#1155] [Q&A]
On our project we have a configuration in which we need to do some testing with RF signals to verify our antenna design works in our situation. For this setup we need to set the TxPower value and The Spreadingfactor value dynamicly but without adr.

is it possible to do something like this within the LoraMacNode code? 
something like `SetTxPower(14);` or something. 

Thanks,
oetze

**Comment by Pirakutu** [comment]
You can do this by exposing the datarate and power you want to use for a certain message, which can be done with mib requests for example. For datarates this is fairly easy, for power you might want to play with the antenna gain, as the power itself is cutoff to an internal regional max eirp value.
